feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/Sessions.kt

```diff
@@ -182,13 +191,60 @@ fun SessionsList(
     ) { dayIndex ->
         val day = days[dayIndex]
         val timetable = scheduleState.schedule.dayToTimetable[day].orEmptyContents()
-        SessionList(timetable) { timetableItem, isFavorited ->
-            SessionListItem(
-                timetableItem = timetableItem,
-                isFavorited = isFavorited,
-                onTimetableClick = onTimetableClick,
-                onFavoriteClick = onFavoriteClick
-            )
+        var currentStartTime = ""
+        SessionList(timetable) { timetableItem, isFavorited, index ->
+            val startLocalDateTime = timetableItem.startsAt
+                .toLocalDateTime(TimeZone.of("UTC+9"))
+            val endLocalDateTime = timetableItem.endsAt
+                .toLocalDateTime(TimeZone.of("UTC+9"))
+            val startTime = LocalTime(
+                hour = startLocalDateTime.hour,
+                minute = startLocalDateTime.minute
+            ).toString()
+            val endTime = LocalTime(
+                hour = endLocalDateTime.hour,
+                minute = endLocalDateTime.minute
+            ).toString()
+            var displayTime = true
+            if (index == 0) currentStartTime = startTime
+            else if (index > 0 && startTime == currentStartTime) {
+                displayTime = false
+            } else {
+                currentStartTime = startTime
```

Comment:
I'm concerned that this can cause unexpected behavior because I think the lambda isn't always called sequentially.

How about making the list in advance? 🩹 

```kotlin
val timeHeaderAndTimetableItems: List<Pair<String?, TimetableItemWithFavorite>> = remember(timetable) {
  ... 
}
SessionList(timeHeaderAndTimetableItems) { (timeHeader, timetableItemWithFavorite), index ->
```